client/src/components/MainContent.js

```diff
@@ -0,0 +1,114 @@
+import { ClipLoader } from 'react-spinners';
+import { chartColors, getProvisionStateInfo } from '../js/ProvisionStateChartUtil';
+import rd3 from 'rd3';
+import FontAwesome from 'react-fontawesome';
+import 'font-awesome/css/font-awesome.css';
+import '../css/style.css';
+
+const PieChart = rd3.PieChart;
+
+/**
+ * Main component to display the application content.
+ */
+class MainContent extends Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			computerSystems: [],
+			isLoading: false,
+			isRowSelected: false,
+			row: null,
+			updateSelectedRow: false
+		};    
+	}
+
+	/**
+	 * Requests the server to grab the computer systems list.
+	 */
+	componentDidMount() {
+		this.setState({isLoading: true});
+
+		fetchAllComputerSystems(
+			(data) => this.setState({status: 'success', computerSystems: data[0].systems, isLoading: false}),
+			(error) => this.setState({status: 'error', isLoading: false}));
+	}
+	
+	/**
+	 * Handle the selection of a row in the ComputerSysDataTable component. By transfering this 
+	 * data we can then load the ComputerSysContainer component.
+	 */
+	onRowSelectCallback = (isRowSelected, row) => {
+		this.setState({
+			isRowSelected: isRowSelected,
+			row: row,
+			updateSelectedRow: false
+		});
+	}
+
+	handleData(data) {
+		const results = JSON.parse(data);
+		if (this.state.isRowSelected) {
+			this.setState({computerSystems: results[0].systems, updateSelectedRow: true});
+		} else {
+			this.setState({computerSystems: results[0].systems, updateSelectedRow: false});
+		}
+	}
+
+	render() {
+		const {status, computerSystems, isLoading, updateSelectedRow} = this.state;
+
+		if (isLoading) {
+			return (
+				<Grid bsClass="main-content">
+					<Row>
+						<Col className="loader">
+							<ClipLoader color={'#34b7b6'} size={100} loading={isLoading} />
+						</Col>
+					</Row>
+				</Grid>
+			);
+		}
+		if (status === 'error') {
+			return (
+				<Grid bsClass="main-content">
+					<Row>
+						<Col className="loader">
+							<FontAwesome className="loader-error" name='exclamation-circle' size='5x' />
+							<br/>
+							<strong>Oops! Something went wrong.</strong>
+						</Col>
+					</Row>
+				</Grid>
+			);
+		}
+
+		return (
+			<Grid bsClass="main-content">
+				<Websocket url={WS_URL} onMessage={this.handleData.bind(this)}/>
+				<Row>
+					<Col className="provision-state-chart">
+						<PieChart data={getProvisionStateInfo(computerSystems)} width={450} height={400} radius={150} innerRadius={90} sectorBorderColor="white" colors={chartColors} />
+					</Col>
+				</Row>
+				<Row>
+					<Col mdOffset={1} md={10}>
+						<ComputerSysDataTable data={computerSystems} callbackFromParent={this.onRowSelectCallback} updateSelectedRow={updateSelectedRow} />
+					</Col>
+				</Row>
+				<Row>
+					<Col mdOffset={1} md={10}>
+						{this.state.isRowSelected ? <ComputerSysContainer data={this.state.row} /> : <strong>Select a computer system from the above table.</strong>}
```

Comment:
True. That is one problem. My first solution was to show the additional information on a hidden line, that will show/hide when a user select one of the rows. But after the implementation I didn't like it because the  datatable was getting to crowded.

To solve this I thought of the solution I implemented. What I believe is that there are two major operation users here (at least): comparation of servers, and see servers details. For the first one, the datatable by it selft works good, we can paginate and sort, and see the major information about each other. For the second one, we don't need many servers to be shown at the datatable and we this the scrolls will not be that long. However, that could still be a problem because it's possible to have a large list and them just what to see a server caracterists and need to scroll down, even if it is just to change the pagination.

In that situation I see two ways to fix that. First, open the detailed information on another page. I personally don't like this solution. The other one, would be to have an go to bottom (or go to the next section) button to lead us right to the next section of the page.

So resuming to fix this situation I would go with the last approach I specify.